### PR TITLE
C++20

### PR DESCRIPTION
**Pull request by petersteneteg**
Switch to C++20 as a requirement, Require Qt6, remove all qt5 related stuff, remove tcb::span use std::span, deprecation fixes. etc.

### CODE REVIEW

File: modules/base/include/modules/base/properties/gaussianproperty.h [hunk 1]
```diff
@@ -38,6 +38,7 @@
 #include <cmath>        // for exp, M_PI
```
Comment (martinfalk) [inline modules/base/include/modules/base/properties/gaussianproperty.h:38]: ;-)
```suggestion
#include <cmath>        // for exp
```

File: modules/qtwidgets/src/angleradiuswidget.cpp [hunk 2]
```diff
@@ -30,8 +30,9 @@
 #include <modules/qtwidgets/angleradiuswidget.h>
 
 #include <algorithm>  // for max, min
-#include <cmath>      // for atan2, cos, sin, sqrt, M_PI
+#include <cmath>      // for atan2, cos, sin, sqrt, std::numbers::pi
 #include <sstream>    // for stringstream, basic_ostream, fixed
+#include <numbers>
```
Comment (martinfalk) [inline modules/qtwidgets/src/angleradiuswidget.cpp:35]:
```suggestion
#include <cmath>      // for atan2, cos, sin, sqrt
#include <sstream>    // for stringstream, basic_ostream, fixed
#include <numbers>    // std::numbers::pi
```

**Review by martinfalk — DISMISSED** [review]

File: modules/qtwidgets/src/angleradiuswidget.cpp (hunk 2 above)
Comment (petersteneteg) [reply modules/qtwidgets/src/angleradiuswidget.cpp:35]: Will rerun the include-what-you-use periodically to fix those things...

File: src/core/util/systemcapabilities.cpp [hunk 3]
```diff
@@ -71,121 +62,25 @@ void SystemCapabilities::retrieveDynamicInfo() {
 const util::BuildInfo& SystemCapabilities::getBuildInfo() const { return buildInfo_; }
 
 bool SystemCapabilities::lookupOSInfo() {
-#ifdef IVW_USE_SIGAR
```
Comment (ResearchDaniel) [inline src/core/util/systemcapabilities.cpp:74]: We should remove these functions unless we replace the functionality. Will be misleading to be able to request the info without it ever being available.

File: cmake/globalconfig.cmake [hunk 4]
```diff
@@ -42,26 +42,25 @@ endif()
...
 else()
-    CHECK_CXX_COMPILER_FLAG("-std=c++17" compiler_supports_cxx17)
+    CHECK_CXX_COMPILER_FLAG("-std=c++20" compiler_supports_cxx20)
 endif()
-if(compiler_supports_cxx17)
-    message(STATUS "C++17 enabled")
-    set(CMAKE_CXX_STANDARD 17 CACHE STRING "C++ ISO Standard" FORCE)
-else()
+if(NOT compiler_supports_cxx20)
     message(FATAL_ERROR "The compiler ${CMAKE_CXX_COMPILER} has no C++17 support. "
```
Comment (ResearchDaniel) [inline cmake/globalconfig.cmake]:
```suggestion
    message(FATAL_ERROR "The compiler ${CMAKE_CXX_COMPILER} has no C++20 support. "
```

File: include/inviwo/qt/editor/fileassociations.h [hunk 5]
```diff
@@ -118,11 +118,7 @@ class IVW_QTEDITOR_API FileAssociations : public QAbstractNativeEventFilter {
     friend FileAssociationData;
 
 public:
-#if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
-    using ResultType = long;
-#else
     using ResultType = qintptr;
```
Comment (ResearchDaniel) [inline include/inviwo/qt/editor/fileassociations.h:121]: No need for this anymore? Can just use the type directly

File: src/qt/CMakeLists.txt [hunk 6]
```diff
@@ -1,24 +1,21 @@
...
-    if(NOT Qt${QT_VERSION_MAJOR}Core_FOUND) 
+    find_package(Qt6 REQUIRED COMPONENTS Core QUIET)
+    if(NOT Qt6Core_FOUND) 
         message(FATAL_ERROR 
             "The inviwo app requiers Qt, we were not able to find it. Consider adding Qt to the "
             "CMAKE_PREFIX_PATH, i.e. start cmake with the option -DCMAKE_PREFIX_PATH=\"/path/to/qt\"")
     endif()
 
     # Create a qt.conf file in the same dir as the executable to help qt find its plugins
     # http://doc.qt.io/qt-5/qt-conf.html
```
Comment (ResearchDaniel) [inline src/qt/CMakeLists.txt]: Is this more correct now?
```suggestion
    # http://doc.qt.io/qt-6/qt-conf.html
```

**Review by ResearchDaniel — DISMISSED** [review]
Looks good!
Could you write a note in the changelog about the update and also include info about common search-and-replace ops, e.g, replace static_cast<float>(M_PI) with std::numbers::pi_v<float>()

File: src/core/util/systemcapabilities.cpp (hunk 3 above)
Comment (petersteneteg) [reply src/core/util/systemcapabilities.cpp:74]: mmm removed SystemCapabilities

File: cmake/globalconfig.cmake (hunk 4 above)
Comment (petersteneteg) [reply cmake/globalconfig.cmake]: fixed

File: include/inviwo/qt/editor/fileassociations.h (hunk 5 above)
Comment (petersteneteg) [reply include/inviwo/qt/editor/fileassociations.h:121]: yes

File: src/qt/CMakeLists.txt (hunk 6 above)
Comment (petersteneteg) [reply src/qt/CMakeLists.txt]: mmm

**Review by martinfalk — DISMISSED** [review]

**Review by martinfalk — DISMISSED** [review]

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,12 @@
 Here we document changes that affect the public API or changes that needs to be communicated to other developers. 
 
+## 2023-03-06 C++20 and Qt6
+A C++20 compiler is now required, in practice, only features that are supported by MSVC, Clang, GCC, and AppleClang are used, which usually means that everything that is supported by the latest version of XCode can bu used.
+Notably ranges is still missing in AppleClang. All support for Qt5 is not removed and Qt6 is now required.
```
Comment (martinfalk) [inline CHANGELOG.md]:
```suggestion
Notably ranges is still missing in AppleClang. Qt6 is now required and all support for Qt5 is now removed. 
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,12 @@
 Here we document changes that affect the public API or changes that needs to be communicated to other developers. 
 
+## 2023-03-06 C++20 and Qt6
+A C++20 compiler is now required, in practice, only features that are supported by MSVC, Clang, GCC, and AppleClang are used, which usually means that everything that is supported by the latest version of XCode can bu used.
+Notably ranges is still missing in AppleClang. All support for Qt5 is not removed and Qt6 is now required. 
+We previously used tcb/span.hpp, that is now replaced with std::span and the dependency has been removed.
+Sigar has also been removed since it has not been updated for over 10 years.
+Unless we find a compelling need for the functionality no replacement will been added.
```
Comment (martinfalk) [inline CHANGELOG.md]:
```suggestion
Unless we find a compelling need for the functionality no replacement will be added.
```

**Review by martinfalk — DISMISSED** [review]

**Review by martinfalk — APPROVED** [review]

**Pull request merged**